api/src/test/java/org/pmiops/workbench/actionaudit/auditors/SumoLogicAuditorTest.java

```diff
@@ -0,0 +1,216 @@
+    sumoLogicAuditor.fireEgressEvent(
+        new EgressEvent()
+            .projectName(EGRESS_EVENT_PROJECT_NAME)
+            .vmName(EGRESS_EVENT_VM_NAME)
+            .timeWindowStart(0l)
+            .egressMib(12.3));
+    verify(mockActionAuditService).send(eventsCaptor.capture());
+    Collection<ActionAuditEvent> events = eventsCaptor.getValue();
+
+    // Ensure all events have the expected set of constant fields.
+    assertThat(events.stream().map(event -> event.getAgentType()).collect(Collectors.toSet()))
+        .containsExactly(AgentType.USER);
+    assertThat(events.stream().map(event -> event.getActionType()).collect(Collectors.toSet()))
+        .containsExactly(ActionType.HIGH_EGRESS);
+    assertThat(events.stream().map(event -> event.getAgentId()).collect(Collectors.toSet()))
+        .containsExactly(USER_ID);
+    assertThat(events.stream().map(event -> event.getAgentEmailMaybe()).collect(Collectors.toSet()))
+        .containsExactly(USER_EMAIL);
+    assertThat(events.stream().map(event -> event.getTargetIdMaybe()).collect(Collectors.toSet()))
+        .containsExactly(WORKSPACE_ID);
+
+    // We should have distinct event rows with values from the egress event.
+    assertThat(
+            events.stream()
+                .filter(
+                    event ->
+                        event.getTargetPropertyMaybe()
+                            == EgressEventTargetProperty.EGRESS_MIB.getPropertyName())
+                .map(event -> event.getNewValueMaybe())
+                .collect(Collectors.toSet()))
+        .containsExactly("12.3");
+    assertThat(
+            events.stream()
+                .filter(
+                    event ->
+                        event.getTargetPropertyMaybe()
+                            == EgressEventTargetProperty.VM_NAME.getPropertyName())
+                .map(event -> event.getNewValueMaybe())
+                .collect(Collectors.toSet()))
+        .containsExactly(EGRESS_EVENT_VM_NAME);
+  }
+
+  @Test
+  public void testNoWorkspaceFound() {
+    exception.expect(BadRequestException.class);
+
+    // When the workspace lookup doesn't succeed, the event is filed w/ a system agent and an
+    // empty target ID.
+    when(mockWorkspaceService.getByNamespace(WORKSPACE_NAMESPACE)).thenReturn(null);
+    sumoLogicAuditor.fireEgressEvent(
+        new EgressEvent().projectName(EGRESS_EVENT_PROJECT_NAME).vmName(EGRESS_EVENT_VM_NAME));
+    verify(mockActionAuditService).send(eventsCaptor.capture());
+    Collection<ActionAuditEvent> events = eventsCaptor.getValue();
+
+    // Some of the properties should be nulled out, since we can't identify the target workspace
+    // for the egress event.
+    assertThat(events.stream().map(event -> event.getAgentId()).collect(Collectors.toSet()))
+        .containsExactly(0);
+    assertThat(events.stream().map(event -> event.getAgentEmailMaybe()).collect(Collectors.toSet()))
+        .containsExactly((Object[]) null);
```

Comment:
This is kind of scary. I'd instead do a  `.filter(Objects::nonNull)` and assert that the set is empty.

Reply:
Good idea – done.